core/src/mindustry/world/Tiles.java

```diff
@@ -27,64 +49,127 @@ public void each(Intc2 cons){
         }
     }
 
-    /** set a tile at a position; does not range-check. use with caution. */
+    /**
+     * Set a tile at a position; does not range-check. use with caution.
+     *
+     * @param x The x location to set the tile at. Make sure it is in bounds with {@link #width}
+     * @param y The y location to set the tile at. Make sure it is in bounds with {@link #width}
+     */
     public void set(int x, int y, Tile tile){
         array[y*width + x] = tile;
     }
 
-    /** @return whether these coordinates are in bounds */
+    /**
+     * Check if coordinates are in bounds of the tile array
+     *
+     * @param x The tile X coordinate
+     * @param y The tile Y coordinate
+     *
+     * @return whether these coordinates are in bounds
+     */
     public boolean in(int x, int y){
         return x >= 0 && x < width && y >= 0 && y < height;
     }
 
-    /** @return a tile at coordinates, or null if out of bounds */
+    /**
+     * Gets a tile at coordinates that can be null if it's out of bounds
+     *
+     * @param x The location on the x-coordinate
+     * @param y The location on the y-coordinate
+     *
+     * @return a tile at coordinates, or null if out of bounds
+     */
     public @Nullable Tile get(int x, int y){
         return (x < 0 || x >= width || y < 0 || y >= height) ? null : array[y*width + x];
     }
 
-    /** @return a tile at coordinates; throws an exception if out of bounds */
+    /**
+     * Get a tile at coordinates with an exception
+     *
+     * @param x The location on the x-coordinate
+     * @param y The location on the y-coordinate
+     *
+     * @return a tile at coordinates
+     * @exception IllegalArgumentException throws this if out of bounds
+     */
     public @NonNull Tile getn(int x, int y){
         if(x < 0 || x >= width || y < 0 || y >= height) throw new IllegalArgumentException(x + ", " + y + " out of bounds: width=" + width + ", height=" + height);
         return array[y*width + x];
     }
 
-    /** @return a tile at coordinates, clamped. */
+    /**
+     * Get a tile at coordinates, and clamps the coordinates if they are out of bounds
+     *
+     * @param x The location on the x-coordinate
+     * @param y The location on the y-coordinate
+     *
+     * @return a tile at coordinates
+     */
     public @NonNull Tile getc(int x, int y){
         x = Mathf.clamp(x, 0, width - 1);
         y = Mathf.clamp(y, 0, height - 1);
         return array[y*width + x];
     }
-
-    /** @return a tile at an iteration index [0, width * height] */
+    /**
+     * Get a tile at an iteration index [0, width * height]
+     *
+     * @param idx The iteration index
+     *
+     * @return a tile at coordinates
+     */
     public @NonNull Tile geti(int idx){
         return array[idx];
     }
 
-    /** @return a tile at an int position (not equivalent to geti) */
+    /**
+     * Get a tile at an int position (not equivalent to geti)
+     *
+     * @param pos The int position
```

Comment:
If you're going to add docs for this, you should explain what an "int position" is: two short x/y coords packed into an int